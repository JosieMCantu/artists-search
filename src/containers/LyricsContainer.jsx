import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import Lyrics from '../components/lyrics/Lyrics'; 
import { fetchLyrics }  from '../LyricService/LyricApi'; 


const LyricsContainer = () => {
  const [loading, setLoading] = useState(true); 
  const [lyrics, setLyrics] = useState(''); 

  const { artistName, title } = useParams();

  useEffect (() => {
    fetchLyrics(artistName, title)
      .then(setLyrics)
      .finally(() => setLoading(false)); 
  }, []);

  if(loading) return <h1>Loading...</h1>; 

  return (
    <div>
      <Lyrics lyrics={lyrics}/>
    </div>
  );
};

export default LyricsContainer; 
