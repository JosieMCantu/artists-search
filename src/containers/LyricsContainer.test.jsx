/**
 * @jest-environment jsdom
 */
import React from 'react'; 
import { render, screen } from '@testing-library/react'; 
import { rest } from 'msw'; 
import { setupServer } from 'msw/node'; 
import LyricsContainer from './LyricsContainer'; 
import mockLyricData from '../../fixtures/mockLyricData.json'; 
import { MemoryRouter } from 'react-router'; 


const server = setupServer(
  // eslint-disable-next-line max-len
  rest.get('https://api.lyrics.ovh/v1/Coldplay/Adventure%20of%20a%20Lifetime', (req, res, ctx) => {
    return res(ctx.json(mockLyricData)); 
  })
); 

describe('Lyrics Container', () => {
  beforeAll(() => server.listen()); 
  afterAll(() => server.close()); 

  // eslint-disable-next-line max-len
  it('displays the lyrics of a song given the artist and title in the parameters', async () => {
    render(<MemoryRouter>
      <LyricsContainer />
    </MemoryRouter>); 

    const lyrics = await screen.findByAltText('lyrics');
    expect(lyrics).toMatchSnapshot();
  }); 
});
