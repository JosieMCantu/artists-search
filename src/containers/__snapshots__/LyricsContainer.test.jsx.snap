// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lyrics Container displays the lyrics of a song given the artist and title in the parameters 1`] = `
<article>
  Turn your magic on, Umi she'd say
Everything you want's a dream away
We are legends, every day
That's what she told me
Turn your magic on, to me she'd say

Everything you want's a dream away

Under this pressure, under this weight

We are diamonds 



I feel my heart beating

I feel my heart underneath my skin

I feel my heart beating

Oh, you make me feel

Like I'm alive again



Alive again!



Oh, you make me feel

Like I'm alive again



Said I can't go on, not in this way

I'm a dream that died by light of day

Gonna hold up half the sky and say

Only I own me



I feel my heart beating

I feel my heart underneath my skin

Oh, I can feel my heart beating

'Cause you make me feel

Like I'm alive again



Alive again!



Oh, you make me feel

Like I'm alive again



Turn your magic on, Umi she'd say

Everything you want's a dream away

Under this pressure, under this weight

We are diamonds taking shape

We are diamonds taking shape



(Woo, woo)



If we've only got this life

This adventure, oh then I

And if we've only got this life

You'll get me through alive

And if we've only got this life

In this adventure, oh then I

Wanna share it with you

With you, with you

I said, oh, say oh



(Woo hoo, woo hoo...)
</article>
`;
